pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -616,20 +615,15 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 		if rss := findSyncStatus(resourceSyncStatus, ssi.Status.Resources); rss == nil || r.needToReapply(resourceSyncStatus, *rss, ssiLog) {
 			// Apply resource
 			ssiLog.Debug("applying resource")
-			var applyResult hiveresource.ApplyResult
-			applyResult, applyErr = h.Apply(resource.Raw)
+
+			applyErr = applyResource(resource, ssiLog, h)
```

Comment:
I don't think it is going to be helpful long-term to embed the unmarshaling into the `applyResource` function. We need to do the same unmarshaling right before in order to get the info to fill out the `SyncStatus`.

Reply:
I have moved this up with the info parsing, this helped simplify the error handling a little and presumably will help if there's a move to get the info from the unstructured in the future.